### DISCUSSION TITLE
CSS doesn't get applied propery when using navigateTo()

### DISCUSSION THREAD

**Discussion by cosbgn** [nuxt/nuxt#16203] [Questions]
If you visit `https://site.com/account` you will be redirected to `https://site.com/auth` by a middleware. 
However the css of the `/auth` page is not applied properly, I'm not sure why. If you reload the page you will see that the correct css get's added only after the reload. (You can see properly the difference on a desktop view). 

What could be the reason of this issue? I use tailwind with JIT. But I don't think it's related.

**Comment by danielroe** [comment]
The issue here is that the initial CSS (loaded on server) is meant to match the page rendered.

In general you should not redirect on client-side before hydration completes or you will encounter mismatch issues like this one.

**Comment by cosbgn** [comment]
I'm a bit confused. The redirect is done by a middleware, like this:

```js 

// /middlewares/auth.js

export default defineNuxtRouteMiddleware(async (to, from) => {
	const loggedIn = useLoggedIn()
	if (!loggedIn.value){
            return navigateTo('/auth')
        }
})
```
How can I wait that hydration is completed? Isn't the best place to check for auth in a middleware like this?

**Reply by danielroe** [reply]
The issue is that your server state is showing 'logged out' and your client is 'logged in'. And before the page is ready, you're navigating away to a different page from the one the server rendered. Either:

1. make the server render the correct page or redirect server-side. <- preferred
2. wait until the app is initialised before redirecting.

**Reply by cosbgn** [reply]
Ok, I guess I'll wait for `nuxt-auth` and once out migrate to it. It will most likely take care of these sort of issues for me.  Thanks!

**Reply by TheAlexLichter** [reply]
PS: Feel free to try out e.g. https://github.com/sidebase/nuxt-auth ☺️
